demo/actions.py

```diff
@@ -693,24 +700,28 @@ def run(self, dispatcher, tracker, domain):
             last_user_event = get_last_event_for(tracker, "user", skip=2)
             if last_user_event:
                 search_text = last_user_event.get("text")
+            else:
+                dispatcher.utter_message(text="Sorry, I can't answer your question.")
+                return []
 
         # Search of docs pages
         algolia = AlgoliaAPI(
             config.algolia_app_id, config.algolia_search_key, config.algolia_docs_index
         )
         alg_res = algolia.search(search_text)
 
-        doc_list = algolia.get_algolia_link(alg_res.get("hits"), 0)
-        doc_list += (
-            "\n" + algolia.get_algolia_link(alg_res.get("hits"), 1)
-            if len(alg_res.get("hits")) > 1
-            else ""
-        )
+        if alg_res:
```

Comment:
If not alg_res, we shouldn't return [], cause it will say nothing and then ask if that helped. Ideally we'd jump straight to the forum search

Reply:
I fixed this as well, by introducing a new slot of type boolean `docs_found`, and used these to distinquish the stories for technical_question intent. If docs are found, it behaves as before. If no docs are found, it will now automatically go and do the forum search.